core/src/epicli/data/common/ansible/playbooks/roles/applications/tasks/main.yml

```diff
@@ -1,21 +1,25 @@
 ---
-- name: Wait until the cluster is available
-  shell: kubectl cluster-info
-  environment:
-    KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
-  changed_when: false
-  retries: 50
-  delay: 1
-  register: result
-  until: result is succeeded
-
-- name: Include applications
-  include_tasks: applications/{{ item.name }}/main.yml
-  no_log: true # to hide sensitive data
-  vars:
-    data: "{{ item }}"
-  loop: "{{ specification.applications }}"
+- name: Deploy applications
   when:
-    - specification.applications is defined and specification.applications | length > 0
-    - item.enabled | default(True)
-    - groups['applications'][0] == inventory_hostname
+    - specification.applications is defined
+    - specification.applications | length > 0
+  always:
+    - name: Remove temporary directory (root)
+      file:
+        path: "{{ deployment_tmp_root_dir }}"
+        state: absent
+  block:
+    - name: Wait until cluster is available
+      include_tasks: wait-for-cluster.yml
+
+    - name: Include applications
+      include_tasks: applications/{{ data.name }}/main.yml
+      loop_control:
+        loop_var: data
+      loop: "{{ _enabled_applications }}"
+      vars:
+        # Application must be explicitly enabled to be included
```

Comment:
Initially we haven't `enabled` setting so for backward compatibility I decided to use `item.enabled | default(True)`.
Maybe it would be good to keep this approach and skip only explicitly disabled apps (`enabled: false`).

Reply:
Please check in latest commits. :+1: 1: